---
title: "pyro.differential abundance.aug5.2021"
output: html_document
---

##load packages
```{r}
library(readr)
library(ggplot2)
library(vegan)
library(tidyr)
library(dplyr)
library(phyloseq)
library(microbiome)
library(ANCOMBC)
library(DESeq2)
library(ggpubr)
library(apeglm)
library(reshape2)
library(phylosmith)
```

##read in data
```{r}
##spring----
##data output from qiime2 viewer at species level - assigned to order or lower features with 10 or less across all samples removed
spring_species_filtered_aug16_2021 <- read_csv("spring.species.aug16.2021.csv")
samplenames<-spring_species_filtered_aug16_2021$index
spring_species_filtered_aug16_2021<-spring_species_filtered_aug16_2021[,2:789]
row.names(spring_species_filtered_aug16_2021)<-samplenames
View(spring_species_filtered_aug16_2021)

#substract reads that were present in the negative control
asv.spring_species.filtered_aug16_2021<-as.data.frame(t(spring_species_filtered_aug16_2021))
which(asv.spring_species.filtered_aug16_2021$neg > 0) ## determine which ASVs the negative has reads in

spring_species_filtered_aug16_2021[,52]<-spring_species_filtered_aug16_2021[,52]-2 #remove the same number of reads from every sample
spring_species_filtered_aug16_2021[,52]<-ifelse(spring_species_filtered_aug16_2021[,52]<0,0,as.numeric(spring_species_filtered_aug16_2021$`k__Fungi;p__Ascomycota;c__Dothideomycetes;o__Pleosporales;f__Didymellaceae;g__Didymella;s__Didymella_exigua`)) #ensure no ASVs had a negative number of reads

#repeat with other ASV
spring_species_filtered_aug16_2021[,314]<-spring_species_filtered_aug16_2021[,314]-4
spring_species_filtered_aug16_2021[,314]<-ifelse(spring_species_filtered_aug16_2021[,314]<0,0,as.numeric(spring_species_filtered_aug16_2021$`k__Fungi;p__Ascomycota;c__Pezizomycetes;o__Pezizales;f__Pyronemataceae;g__Geopyxis;s__Geopyxis_carbonaria`))

row.names(spring_species_filtered_aug16_2021)<-samplenames #rename the rows

asv.spring_species.filtered_aug16_2021<-as.data.frame(t(spring_species_filtered_aug16_2021))
which(asv.spring_species.filtered_aug16_2021$neg > 0) ## check that it worked

##seperate out the metadata: sample-type, sample-date, severity
meta.data.spring<-spring_species_filtered_aug16_2021[,786:788]
meta.data.microplots.spring<-spring_species_filtered_aug16_2021[c(1:37,46:94),786:788]
samplenames.microplots<-samplenames[c(1:37,46:94)]

##reorder severity levels 
meta.data.microplots.spring$Severity<-ordered(meta.data.microplots.spring$Severity,levels=c("unburnt","surface","medium","med-severe","severe","lp-severe"))
meta.data.microplots.spring$sample_date<-ordered(meta.data.microplots.spring$sample_date,levels=c("sept","oct","may","june"))

#rename factors 
meta.data.microplots.spring$Severity <- factor(meta.data.microplots.spring$Severity, labels = c("Unburnt","Low","Medium","Medium-Severe","Severe","LP-Severe"))
levels(meta.data.microplots.spring$Severity)
meta.data.microplots.spring$sample_date <- factor(meta.data.microplots.spring$sample_date, labels = c("September","October","May","June"))
levels(meta.data.microplots.spring$sample_date)
meta.data.microplots.spring$sample_type<-factor(meta.data.microplots.spring$sample_type,labels = c("Mineral","Surface"))
levels(meta.data.microplots.spring$sample_type)

View(meta.data.microplots.spring)

###Fall----
Taxa <- read_csv("taxa-bar-plots-order-10.may30.2022.csv")
samplenames.fall<-Taxa$index
Taxa<-as.data.frame(Taxa[,2:ncol(Taxa)])
rownames(Taxa)<-samplenames.fall
Taxa<-Taxa[c(-16,-18,-44),] #this is to remove samples O15C which was rerun successfully in the spring
View(Taxa)

#seperate out the metadata: sample-type, sample-date, severity
meta.data.fall<-Taxa[,691:692]
meta.data.fall<-separate(meta.data.fall,col="sample-type",into=c("sample_type","sample_date"),sep="-")
meta.data.microplots.fall<-meta.data.fall[c(1:27,31:61),] 
colnames(meta.data.fall)<-c("sample_type","sample_date","Severity")
colnames(meta.data.microplots.fall)<-c("sample_type","sample_date","Severity")
View(meta.data.microplots.fall)

#order
meta.data.microplots.fall$Severity<-ordered(meta.data.microplots.fall$Severity,levels=c("unburnt","surface","medium","med-severe","severe"))
meta.data.microplots.fall$sample_date<-ordered(meta.data.microplots.fall$sample_date,levels=c("sept","oct"))

#rename factors 
meta.data.microplots.fall$Severity <- factor(meta.data.microplots.fall$Severity, labels = c("Unburnt","Low","Medium","Medium-Severe","Severe"))
levels(meta.data.microplots.fall$Severity)
meta.data.microplots.fall$sample_date <- factor(meta.data.microplots.fall$sample_date, labels = c("September","October"))
levels(meta.data.microplots.fall$sample_date)
meta.data.microplots.fall$sample_type<-factor(meta.data.microplots.fall$sample_type,labels = c("Mineral","Surface"))
levels(meta.data.microplots.fall$sample_type)

```

#create phyloseq objects
```{r}
##spring ----
##modify data sheet to fit format - taxa must be rows 
species.data.otu.spring<-t(spring_species_filtered_aug16_2021[,1:785])
View(species.data.otu.spring)

##create the species data 
phylo.species.spring<-otu_table(species.data.otu.spring,taxa_are_rows = TRUE)
taxa_names(phylo.species.spring)<-colnames(spring_species_filtered_aug16_2021[,1:785])
sample_names(phylo.species.spring)<-rownames(spring_species_filtered_aug16_2021)

#create taxonomy table in OTU format
taxa.spring<-as.data.frame(colnames(spring_species_filtered_aug16_2021[,1:785]))
View(taxa.spring)
colnames(taxa.spring)<-"full.taxa"
taxa.spring<-separate(taxa.spring,col= "full.taxa",into = c("Kingdom","Phylum","Class","Order","Family","Genus","Species"),sep = ";")

phylo.taxonomy.spring<-tax_table(taxa.spring)
phylo.taxonomy.spring
taxa_names(phylo.taxonomy.spring)<-colnames(spring_species_filtered_aug16_2021[,1:785])

#make sample meta data
phylo.sample.meta.spring<-sample_data(meta.data.spring)
sample_names(phylo.sample.meta.spring)<-row.names(spring_species_filtered_aug16_2021)


#now merge species, taxonomy, and metadata
phylo.spring<-phyloseq(phylo.species.spring,phylo.taxonomy.spring,phylo.sample.meta.spring)
phylo.spring

# redo but without mock, morel, neg samples
microplot.data.otu.spring<-species.data.otu.spring[1:785,c(1:37,46:94)]
microplot.taxa.spring<-otu_table(microplot.data.otu.spring,taxa_are_rows = TRUE)
phylo.sample.meta.spring.microplots<-sample_data(meta.data.microplots.spring)
sample_names(microplot.taxa.spring)<-samplenames.microplots
sample_names(phylo.sample.meta.spring.microplots)<-samplenames.microplots

phylo.spring.2<-phyloseq(microplot.taxa.spring,phylo.taxonomy.spring,phylo.sample.meta.spring.microplots)


##Fall----
##modify data sheet to fit format - taxa must be rows 
species.data.otu.fall<-t(Taxa[,1:690])
View(species.data.otu.fall)

##create species data
phylo.taxa.fall<-otu_table(species.data.otu.fall,taxa_are_rows = TRUE)
phylo.taxa.fall

#create taxonomy table 
taxa.fall<-as.data.frame(colnames(Taxa[,1:690]))
View(taxa.fall)
colnames(taxa.fall)<-"full.taxa"
taxa.fall<-separate(taxa.fall,col= "full.taxa",into = c("Kingdom","Phylum","Class","Order","Family","Genus","Species"),sep = ";")
##warning here is just because some do not go past order level 

phylo.taxonomy.fall<-tax_table(taxa.fall)
taxa_names(phylo.taxonomy.fall)<-colnames(Taxa[,1:690])

#make sample data
phylo.sample.meta.fall<-sample_data(meta.data.fall)
phylo.sample.meta.fall
sample_names(phylo.sample.meta.fall)<-row.names(Taxa)
sample_names(phylo.taxa.fall)<-row.names(Taxa)

#now merge
phylo.fall<-phyloseq(phylo.taxa.fall,phylo.taxonomy.fall,phylo.sample.meta.fall)
phylo.fall

# redo but without mock samples
microplot.data.otu.fall<-as.data.frame(Taxa[c(1:27,31:61),1:690])
microplot.data.otu.fall<-t(microplot.data.otu.fall)
microplot.taxa.fall<-otu_table(microplot.data.otu.fall,taxa_are_rows = TRUE)
phylo.sample.meta.fall.microplots<-sample_data(meta.data.microplots.fall)
sample_names(phylo.sample.meta.fall.microplots)<-samplenames.fall[c(-16,-18,-30,-31,-32,-44)]
sample_names(microplot.taxa.fall)<-samplenames.fall[c(-16,-18,-30,-31,-32,-44)]
taxa_names(phylo.taxonomy.fall)<-colnames(Taxa[c(1:27,31:61),1:690])

phylo.fall.2<-phyloseq(microplot.taxa.fall,phylo.taxonomy.fall,phylo.sample.meta.fall.microplots)

#clean out taxa that were only present in mock... 
alltaxa.master<-otu_table(master.phylo)
goodtaxa.master<-alltaxa.master[-which(rowSums(otu_table(master.phylo)) == 0),]
goodtaxa.master<-otu_table(goodtaxa.master,taxa_are_rows = TRUE)
taxa.good.master<-as.data.frame(rownames(goodtaxa.master))
colnames(taxa.good.master)<-"full.taxa"
taxa.good.master<-separate(taxa.good.master,col= "full.taxa",into = c("Kingdom","Phylum","Class","Order","Family","Genus","Species"),sep = ";")
good.taxonomy.master<-tax_table(taxa.good.master)
taxa_names(good.taxonomy.master)<-rownames(goodtaxa.master)
sd.master<-sample_data(master.phylo)
master.phylo<-merge_phyloseq(sd.master,goodtaxa.master,good.taxonomy.master)
master.phylo

###now merge spring and fall ----
master.phylo<-merge_phyloseq(phylo.spring.2,phylo.fall.2)
sample_data(master.phylo)

```


##using ANCOM-BC to test differential abun of pyrophilous species
```{r}
#set.seed(15)
#dif.abun<-ancombc(genus_data,formula="Severity",p_adj_method = "fdr",lib_cut = 100,group = "Severity") # comparing the different severity levels 
#View(dif.abun$res$diff_abn)

#try comparing only burnt to unburnt 
# create metadata variable where severe, med-severe, medium, surface all = burnt 
meta.data.microplots.fall$burn.status<-ifelse(meta.data.microplots.fall$Severity=="Severe","Burnt",if_else(meta.data.microplots.fall$Severity=="Medium-Severe","Burnt",if_else(meta.data.microplots.fall$Severity=="Medium","Burnt",if_else(meta.data.microplots.fall$Severity=="Low","Burnt",if_else(meta.data.microplots.fall$Severity=="Unburnt","Unburnt",NA_character_)))))

meta.data.microplots.spring$burn.status<-ifelse(meta.data.microplots.spring$Severity=="LP-Severe","Burnt",ifelse(meta.data.microplots.spring$Severity=="Severe","Burnt",if_else(meta.data.microplots.spring$Severity=="Medium-Severe","Burnt",if_else(meta.data.microplots.spring$Severity=="Medium","Burnt",if_else(meta.data.microplots.spring$Severity=="Low","Burnt",if_else(meta.data.microplots.spring$Severity=="Unburnt","Unburnt",NA_character_))))))

##read in new meta.data to phyloseq
phylo.sample.meta.fall.status<-sample_data(meta.data.microplots.fall)
phylo.sample.meta.spring.status<-sample_data(meta.data.microplots.spring)
sample_names(phylo.sample.meta.spring.status)<-sample_names(phylo.sample.meta.spring.microplots)
sample_names(phylo.sample.meta.fall.status)<-sample_names(phylo.sample.meta.fall.microplots)
phylo.fall.2<-phyloseq(microplot.taxa.fall,phylo.taxonomy.fall,phylo.sample.meta.fall.status)
phylo.spring.2<-phyloseq(microplot.taxa.spring,phylo.taxonomy.spring,phylo.sample.meta.spring.status)

master.phylo<-merge_phyloseq(phylo.spring.2,phylo.fall.2)
sample_data(master.phylo)

##lets test at the genus level
genus_data<-aggregate_taxa(master.phylo,"ta6")
genus_data.nolp<-subset_samples(genus_data, Severity != "LP-Severe")

#re-do ancombc with burn status
set.seed(14)
dif.abun.status<-ancombc(genus_data,formula="burn.status+sample_date+sample_type",p_adj_method = "bonferroni",lib_cut = 100,group = "burn.status",global = TRUE)
View(dif.abun.status$res$beta)

dif.abun.status.nolp<-ancombc(genus_data.nolp,formula="burn.status+sample_date+sample_type",p_adj_method = "bonferroni",lib_cut = 100,group = "burn.status",global = TRUE)
View(dif.abun.status.nolp$res$beta)

#do ancombc with date 
dif.abun.date<-ancombc(genus_data.nolp,formula="burn.status+sample_date+sample_type",p_adj_method = "bonferroni",lib_cut = 100,group = "sample_date",global = TRUE)
View(dif.abun.date$res$beta)

#try ancombc for the date but without the unburnt data
#genus_data_burn<-subset_samples(genus_data.nolp, Severity != "Unburnt") #remove unburnt samples from data set
#sample_data(genus_data_burn)
#dif.abun.date.2<-ancombc(genus_data_burn,formula="sample_date+sample_type",p_adj_method = "bonferroni",lib_cut = 100,group = "sample_date",global = TRUE)
#View(dif.abun.date.2$res$beta)
#removed Severity as part of the formula
```

##plotting results of burn status
```{r}
dif.abun.df<-as.data.frame(dif.abun.status$res$beta[,1])
View(dif.abun.df)
col.names<-c("status(unburnt-burnt)")
colnames(dif.abun.df)<-col.names

df_fig1 = data.frame(dif.abun.df * dif.abun.status$res$diff_abn[,1], check.names = FALSE) 
df_fig1$taxon_id<-rownames(dif.abun.status$res$beta)
df_fig2 = data.frame(dif.abun.status$res$se[,1] * dif.abun.status$res$diff_abn[,1], check.names = FALSE) 
colnames(df_fig2)<-col.names
df_fig2$taxon_id<-rownames(dif.abun.status$res$beta)
df_fig = left_join(df_fig1,df_fig2,by="taxon_id")
df_fig$taxon_id = factor(df_fig$taxon_id, levels = df_fig$taxon_id)
#changing taxon to only genus name showing
df_fig<-separate(df_fig,col= "taxon_id",into = c("higher","Genus"),sep = "g_")
df_fig<-df_fig[,-2]
df_fig.plus<-filter(df_fig, df_fig$`status(unburnt-burnt).x`>0)
df_fig.minus<-filter(df_fig, df_fig$`status(unburnt-burnt).x`<0)
df_fig<-rbind(df_fig.plus,df_fig.minus)
df_fig<-df_fig[-which(df_fig$Genus=="_unidentified"),]
df_fig<-df_fig[-is.na(df_fig$Genus),]
p = ggplot(data = df_fig, 
           aes(x = reorder(Genus,-`status(unburnt-burnt).x`), y = `status(unburnt-burnt).x`)) + 
  geom_bar(stat = "identity", width = 0.7, 
           position = position_dodge(width = 0.4)) +
  geom_errorbar(aes(ymin = `status(unburnt-burnt).x` - `status(unburnt-burnt).y`, ymax = `status(unburnt-burnt).x` + `status(unburnt-burnt).y`), width = 0.2,
                position = position_dodge(0.05), color = "black") + 
  labs(x = NULL, y = "Log fold change", 
       title = "Differential abundance between Unburnt and Burnt") + 
  theme_bw() + 
  theme(legend.position = "none",
        plot.title = element_text(hjust = 0.5),
        panel.grid.minor.y = element_blank(),
        axis.text.x = element_text(angle = 60, hjust = 1))
p

nrow(df_fig.minus) #79
nrow(df_fig.plus) #47

dif.abun.df<-as.data.frame(dif.abun.status.nolp$res$beta[,1])
View(dif.abun.df)
col.names<-c("status(unburnt-burnt)")
colnames(dif.abun.df)<-col.names

df_fig1 = data.frame(dif.abun.df * dif.abun.status.nolp$res$diff_abn[,1], check.names = FALSE) 
df_fig1$taxon_id<-rownames(dif.abun.status.nolp$res$beta)
df_fig2 = data.frame(dif.abun.status.nolp$res$se[,1] * dif.abun.status.nolp$res$diff_abn[,1], check.names = FALSE) 
colnames(df_fig2)<-col.names
df_fig2$taxon_id<-rownames(dif.abun.status.nolp$res$beta)
df_fig = left_join(df_fig1,df_fig2,by="taxon_id")
df_fig$taxon_id = factor(df_fig$taxon_id, levels = df_fig$taxon_id)
#changing taxon to only genus name showing
df_fig<-separate(df_fig,col= "taxon_id",into = c("higher","Genus"),sep = "g_")
df_fig<-df_fig[,-2]
df_fig.plus<-filter(df_fig, df_fig$`status(unburnt-burnt).x`>0)
df_fig.minus<-filter(df_fig, df_fig$`status(unburnt-burnt).x`<0)
df_fig<-rbind(df_fig.plus,df_fig.minus)
df_fig<-df_fig[-which(df_fig$Genus=="_unidentified"),]
df_fig<-df_fig[-is.na(df_fig$Genus),]
q = ggplot(data = df_fig, 
           aes(x = reorder(Genus,-`status(unburnt-burnt).x`), y = `status(unburnt-burnt).x`)) + 
  geom_bar(stat = "identity", width = 0.7, 
           position = position_dodge(width = 0.4)) +
  geom_errorbar(aes(ymin = `status(unburnt-burnt).x` - `status(unburnt-burnt).y`, ymax = `status(unburnt-burnt).x` + `status(unburnt-burnt).y`), width = 0.2,
                position = position_dodge(0.05), color = "black") + 
  labs(x = NULL, y = "Log fold change", 
       title = "Differential abundance between Unburnt and Burnt") + 
  theme_bw() + 
  theme(legend.position = "none",
        plot.title = element_text(hjust = 0.5),
        panel.grid.minor.y = element_blank(),
        axis.text.x = element_text(angle = 60, hjust = 1))
q

nrow(df_fig.minus) #77
nrow(df_fig.plus) #49

##no big difference if I do or do not include the LP-Sev sites 

```

#condense plotting to only include taxa with relative abundance greater than 0.01%
```{r}
gen.propor<-taxa_proportions(genus_data.nolp,"ta6",treatment = "sample")
View(gen.propor)

burnt.gen<-subset_samples(genus_data.nolp, Severity != "Unburnt")
gen.propor.burnt<-taxa_proportions(burnt.gen,"ta6",treatment = "sample")
View(gen.propor.burnt)

pyrophilous.taxa<-unique(df_fig.minus$Genus)
mean(gen.propor[ta6 == "g__Cladosporium",]$Proportion ) #0.77
mean(gen.propor.burnt[ta6 == "g__Cladosporium",]$Proportion ) 
mean(gen.propor[ta6 == "g__Mycosphaerella",]$Proportion ) #1.3
mean(gen.propor.burnt[ta6 == "g__Mycosphaerella",]$Proportion ) 
mean(gen.propor[ta6 == "g__Aureobasidium",]$Proportion ) #0.08
mean(gen.propor.burnt[ta6 == "g__Aureobasidium",]$Proportion ) 
mean(gen.propor[ta6 == "g__Endoconidioma",]$Proportion )  #0.07
mean(gen.propor.burnt[ta6 == "g__Endoconidioma",]$Proportion ) 
mean(gen.propor[ta6 == "g__Hormonema",]$Proportion ) #0.27
mean(gen.propor.burnt[ta6 == "g__Hormonema",]$Proportion ) 
mean(gen.propor[ta6 == "g__Perusta",]$Proportion ) #0.10
mean(gen.propor.burnt[ta6 == "g__Perusta",]$Proportion ) 
mean(gen.propor[ta6 == "g__Coniothyrium",]$Proportion ) #1.47
mean(gen.propor.burnt[ta6 == "g__Coniothyrium",]$Proportion ) 
mean(gen.propor[ta6 == "g__Didymella",]$Proportion ) #5.66
mean(gen.propor.burnt[ta6 == "g__Didymella",]$Proportion ) 
mean(gen.propor[ta6 == "g__Kalmusia",]$Proportion ) #<0.01
mean(gen.propor.burnt[ta6 == "g__Kalmusia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Pleurophoma",]$Proportion ) #<0.01
mean(gen.propor.burnt[ta6 == "g__Pleurophoma",]$Proportion ) 
mean(gen.propor[ta6 == "g__Acericola",]$Proportion ) #<0.01
mean(gen.propor.burnt[ta6 == "g__Acericola",]$Proportion ) 
mean(gen.propor[ta6 == "g__Dematiopleospora",]$Proportion ) #<0.01
mean(gen.propor.burnt[ta6 == "g__Dematiopleospora",]$Proportion ) 
mean(gen.propor[ta6 == "g__Juncaceicola",]$Proportion ) #0.031
mean(gen.propor.burnt[ta6 == "g__Juncaceicola",]$Proportion ) 
mean(gen.propor[ta6 == "g__Leptospora",]$Proportion ) #<0.01
mean(gen.propor.burnt[ta6 == "g__Leptospora",]$Proportion ) 
mean(gen.propor[ta6 == "g__Phaeosphaeria",]$Proportion ) #0.011
mean(gen.propor.burnt[ta6 == "g__Phaeosphaeria",]$Proportion )
mean(gen.propor[ta6 == "g__Pseudoophiobolus",]$Proportion ) #<0.01
mean(gen.propor.burnt[ta6 == "g__Pseudoophiobolus",]$Proportion )
mean(gen.propor[ta6 == "g__Alternaria",]$Proportion ) #0.49
mean(gen.propor.burnt[ta6 == "g__Alternaria",]$Proportion )
mean(gen.propor[ta6 == "g__Comoclathris",]$Proportion ) #0.02
mean(gen.propor.burnt[ta6 == "g__Comoclathris",]$Proportion )
mean(gen.propor[ta6 == "g__Neocamarosporium",]$Proportion ) #<0.01
mean(gen.propor.burnt[ta6 == "g__Neocamarosporium",]$Proportion )
mean(gen.propor[ta6 == "g__Stemphylium",]$Proportion ) #<0.01
mean(gen.propor.burnt[ta6 == "g__Stemphylium",]$Proportion )
mean(gen.propor[ta6 == "g__Helicoma",]$Proportion ) #<0.01
mean(gen.propor.burnt[ta6 == "g__Helicoma",]$Proportion )
mean(gen.propor[ta6 == "g__Fusicladium",]$Proportion ) #2.02
mean(gen.propor.burnt[ta6 == "g__Fusicladium",]$Proportion )
mean(gen.propor[ta6 == "g__Sarcinomyces",]$Proportion ) #0.026
mean(gen.propor.burnt[ta6 == "g__Sarcinomyces",]$Proportion )
mean(gen.propor[ta6 == "g__Penicillium",]$Proportion ) #1.80
mean(gen.propor.burnt[ta6 == "g__Penicillium",]$Proportion )
mean(gen.propor[ta6 == "g__Calyptrozyma",]$Proportion ) #0.48
mean(gen.propor.burnt[ta6 == "g__Calyptrozyma",]$Proportion ) 
mean(gen.propor[ta6 == "g__Fontanospora",]$Proportion ) #0.021
mean(gen.propor.burnt[ta6 == "g__Fontanospora",]$Proportion ) 
mean(gen.propor[ta6 == "g__Rhynchosporium",]$Proportion ) #0.37
mean(gen.propor.burnt[ta6 == "g__Rhynchosporium",]$Proportion ) 
mean(gen.propor[ta6 == "g__Peziza",]$Proportion ) #0.049
mean(gen.propor.burnt[ta6 == "g__Peziza",]$Proportion ) 
mean(gen.propor[ta6 == "g__Anthracobia",]$Proportion ) #1.31
mean(gen.propor.burnt[ta6 == "g__Anthracobia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Geopyxis",]$Proportion ) #4.97
mean(gen.propor.burnt[ta6 == "g__Geopyxis",]$Proportion ) 
mean(gen.propor[ta6 == "g__Tricharina",]$Proportion ) #0.286
mean(gen.propor.burnt[ta6 == "g__Tricharina",]$Proportion )
mean(gen.propor[ta6 == "g__Trichophaea",]$Proportion ) #<0.01
mean(gen.propor.burnt[ta6 == "g__Trichophaea",]$Proportion ) 
mean(gen.propor[ta6 == "g__Warcupia",]$Proportion ) #0.44
mean(gen.propor.burnt[ta6 == "g__Warcupia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Coniochaeta",]$Proportion ) #0.192
mean(gen.propor.burnt[ta6 == "g__Coniochaeta",]$Proportion ) 
mean(gen.propor[ta6 == "g__Neurospora",]$Proportion ) #0.56
mean(gen.propor.burnt[ta6 == "g__Neurospora",]$Proportion ) 
mean(gen.propor[ta6 == "g__Saitoella",]$Proportion ) #0.166
mean(gen.propor.burnt[ta6 == "g__Saitoella",]$Proportion ) 
mean(gen.propor[ta6 == "g__Taphrina",]$Proportion ) #0.20
mean(gen.propor.burnt[ta6 == "g__Taphrina",]$Proportion ) 
mean(gen.propor[ta6 == "g__Coprinopsis",]$Proportion ) #0.015
mean(gen.propor.burnt[ta6 == "g__Coprinopsis",]$Proportion ) 
mean(gen.propor[ta6 == "g__Pachylepyrium",]$Proportion ) #0.0467
mean(gen.propor.burnt[ta6 == "g__Pachylepyrium",]$Proportion ) 
mean(gen.propor[ta6 == "g__Pholiota",]$Proportion ) #0.149
mean(gen.propor.burnt[ta6 == "g__Pholiota",]$Proportion ) 
mean(gen.propor[ta6 == "g__Myxomphalia",]$Proportion ) #0.0180
mean(gen.propor.burnt[ta6 == "g__Myxomphalia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Cotylidia",]$Proportion ) #0.02367
mean(gen.propor.burnt[ta6 == "g__Cotylidia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Cystobasidium",]$Proportion ) #<0.01
mean(gen.propor.burnt[ta6 == "g__Cystobasidium",]$Proportion ) 
mean(gen.propor[ta6 == "g__Mastigobasidium",]$Proportion ) #0.011
mean(gen.propor.burnt[ta6 == "g__Mastigobasidium",]$Proportion ) 
mean(gen.propor[ta6 == "g__Curvibasidium",]$Proportion ) #0.081
mean(gen.propor.burnt[ta6 == "g__Curvibasidium",]$Proportion ) 
mean(gen.propor[ta6 == "g__Rhodosporidiobolus",]$Proportion ) #0.141
mean(gen.propor.burnt[ta6 == "g__Rhodosporidiobolus",]$Proportion ) 
mean(gen.propor[ta6 == "g__Cystofilobasidium",]$Proportion ) #0.022
mean(gen.propor.burnt[ta6 == "g__Cystofilobasidium",]$Proportion ) 
mean(gen.propor[ta6 == "g__Sporobolomyces",]$Proportion ) #0.055
mean(gen.propor.burnt[ta6 == "g__Sporobolomyces",]$Proportion ) 
mean(gen.propor[ta6 == "g__Itersonilia",]$Proportion ) #<0.01
mean(gen.propor.burnt[ta6 == "g__Itersonilia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Phaffia",]$Proportion ) #0.10
mean(gen.propor.burnt[ta6 == "g__Phaffia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Udeniomyces",]$Proportion ) #0.040
mean(gen.propor.burnt[ta6 == "g__Udeniomyces",]$Proportion ) 
mean(gen.propor[ta6 == "g__Filobasidium",]$Proportion ) #0.055
mean(gen.propor.burnt[ta6 == "g__Filobasidium",]$Proportion ) 
mean(gen.propor[ta6 == "g__Naganishia",]$Proportion ) #0.359
mean(gen.propor.burnt[ta6 == "g__Naganishia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Holtermanniella",]$Proportion ) #0.328
mean(gen.propor.burnt[ta6 == "g__Holtermanniella",]$Proportion ) 
mean(gen.propor[ta6 == "g__Bullera",]$Proportion ) #<0.01
mean(gen.propor.burnt[ta6 == "g__Bullera",]$Proportion ) 
mean(gen.propor[ta6 == "g__Dioszegia",]$Proportion ) #0.0434
mean(gen.propor.burnt[ta6 == "g__Dioszegia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Vishniacozyma",]$Proportion ) #0.399
mean(gen.propor.burnt[ta6 == "g__Vishniacozyma",]$Proportion ) 
mean(gen.propor[ta6 == "g__Gelidatrema",]$Proportion ) #<0.01
mean(gen.propor.burnt[ta6 == "g__Gelidatrema",]$Proportion ) 
mean(gen.propor[ta6 == "g__Papiliotrema",]$Proportion ) #0.011
mean(gen.propor.burnt[ta6 == "g__Papiliotrema",]$Proportion ) 
mean(gen.propor[ta6 == "g__Borealophlyctis",]$Proportion ) #0.0285
mean(gen.propor.burnt[ta6 == "g__Borealophlyctis",]$Proportion ) 
mean(gen.propor[ta6 == "g__Powellomyces",]$Proportion ) #0.013
mean(gen.propor.burnt[ta6 == "g__Powellomyces",]$Proportion ) 
mean(gen.propor[ta6 == "g__Spizellomyces",]$Proportion ) #0.567
mean(gen.propor.burnt[ta6 == "g__Spizellomyces",]$Proportion ) 
mean(gen.propor[ta6 == "g__Basidioascus",]$Proportion) #3.79%
mean(gen.propor.burnt[ta6 == "g__Basidioascus",]$Proportion) #5.01%
mean(gen.propor[ta6 == "g__Pyronema",]$Proportion ) #13.69
mean(gen.propor.burnt[ta6 == "g__Pyronema",]$Proportion ) #18.24
mean(gen.propor[ta6 == "g__Morchella",]$Proportion ) #0.56
mean(gen.propor.burnt[ta6 == "g__Morchella",]$Proportion ) #0.72%
mean(gen.propor[ta6 == "g__Comoclathris",]$Proportion ) #0.0264
mean(gen.propor.burnt[ta6 == "g__Comoclathris",]$Proportion ) 
mean(gen.propor[ta6 == "g__Rhodotorula",]$Proportion ) ##<0.01
mean(gen.propor.burnt[ta6 == "g__Rhodotorula",]$Proportion ) 
mean(gen.propor[ta6 == "g__Ramimonilia",]$Proportion ) ##<0.01
mean(gen.propor.burnt[ta6 == "g__Ramimonilia",]$Proportion ) 

sensitive.taxa<-unique(df_fig.plus$Genus)
mean(gen.propor[ta6 == "g__Extremus",]$Proportion ) ##<0.05
mean(gen.propor.burnt[ta6 == "g__Extremus",]$Proportion ) 
mean(gen.propor[ta6 == "g__Meristemomyces",]$Proportion ) ##0.096
mean(gen.propor.burnt[ta6 == "g__Meristemomyces",]$Proportion ) 
mean(gen.propor[ta6 == "g__Sympodiella",]$Proportion ) ##0.1399
mean(gen.propor.burnt[ta6 == "g__Sympodiella",]$Proportion ) 
mean(gen.propor[ta6 == "g__Venturia",]$Proportion ) ##0.1475
mean(gen.propor.burnt[ta6 == "g__Venturia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Cladophialophora",]$Proportion ) ##0.075
mean(gen.propor.burnt[ta6 == "g__Cladophialophora",]$Proportion ) 
mean(gen.propor[ta6 == "g__Knufia",]$Proportion ) ##0.1129
mean(gen.propor.burnt[ta6 == "g__Knufia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Sarcoleotia",]$Proportion ) ##0.57
mean(gen.propor.burnt[ta6 == "g__Sarcoleotia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Cladonia",]$Proportion ) ##0.198
mean(gen.propor.burnt[ta6 == "g__Cladonia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Parmelia",]$Proportion ) ##0.217
mean(gen.propor.burnt[ta6 == "g__Parmelia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Cryptosporiopsis",]$Proportion ) ##<0.05
mean(gen.propor.burnt[ta6 == "g__Cryptosporiopsis",]$Proportion ) 
mean(gen.propor[ta6 == "g__Scleropezicula",]$Proportion ) ##0.0676
mean(gen.propor.burnt[ta6 == "g__Scleropezicula",]$Proportion ) 

mean(gen.propor[ta6 == "g__Leucosporidium",]$Proportion ) ##<0.05
mean(gen.propor.burnt[ta6 == "g__Leucosporidium",]$Proportion ) 
mean(gen.propor[ta6 == "g__Infundichalara",]$Proportion ) ##0.183
mean(gen.propor.burnt[ta6 == "g__Infundichalara",]$Proportion ) 
mean(gen.propor[ta6 == "g__Cadophora",]$Proportion ) ##0.292
mean(gen.propor.burnt[ta6 == "g__Cadophora",]$Proportion ) 
mean(gen.propor[ta6 == "g__Chalara",]$Proportion ) ##0.099
mean(gen.propor.burnt[ta6 == "g__Chalara",]$Proportion ) 
mean(gen.propor[ta6 == "g__Xenopolyscytalum",]$Proportion ) ##<0.05
mean(gen.propor.burnt[ta6 == "g__Xenopolyscytalum",]$Proportion ) 
mean(gen.propor[ta6 == "g__Hyaloscypha",]$Proportion ) ##0.782
mean(gen.propor.burnt[ta6 == "g__Hyaloscypha",]$Proportion ) 
mean(gen.propor[ta6 == "g__Lachnum",]$Proportion ) ##0.129
mean(gen.propor.burnt[ta6 == "g__Lachnum",]$Proportion ) 
mean(gen.propor[ta6 == "g__Pezoloma",]$Proportion ) ##0.240
mean(gen.propor.burnt[ta6 == "g__Pezoloma",]$Proportion ) 
mean(gen.propor[ta6 == "g__Phialocephala",]$Proportion ) ##0.0570
mean(gen.propor.burnt[ta6 == "g__Phialocephala",]$Proportion ) 
mean(gen.propor[ta6 == "g__Gremmenia",]$Proportion ) ##0.0953
mean(gen.propor.burnt[ta6 == "g__Gremmenia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Phacidium",]$Proportion ) ##<0.05
mean(gen.propor.burnt[ta6 == "g__Phacidium",]$Proportion ) 
mean(gen.propor[ta6 == "g__Wilcoxina",]$Proportion ) ##4.836
mean(gen.propor.burnt[ta6 == "g__Wilcoxina",]$Proportion ) 
mean(gen.propor[ta6 == "g__Clavaria",]$Proportion ) ##0.465
mean(gen.propor.burnt[ta6 == "g__Clavaria",]$Proportion ) 
mean(gen.propor[ta6 == "g__Inocybe",]$Proportion ) ##3.09
mean(gen.propor.burnt[ta6 == "g__Inocybe",]$Proportion ) 
mean(gen.propor[ta6 == "g__Mycena",]$Proportion ) ##1.112
mean(gen.propor.burnt[ta6 == "g__Mycena",]$Proportion ) 
mean(gen.propor[ta6 == "g__Amphinema",]$Proportion ) ##0.443
mean(gen.propor.burnt[ta6 == "g__Amphinema",]$Proportion ) 
mean(gen.propor[ta6 == "g__Athelia",]$Proportion ) ##0.0508
mean(gen.propor.burnt[ta6 == "g__Athelia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Rhizopogon",]$Proportion ) ##1.00
mean(gen.propor.burnt[ta6 == "g__Rhizopogon",]$Proportion ) 
mean(gen.propor[ta6 == "g__Pseudotomentella",]$Proportion ) ##0.249
mean(gen.propor.burnt[ta6 == "g__Pseudotomentella",]$Proportion ) 
mean(gen.propor[ta6 == "g__Tomentella",]$Proportion ) ##0.278
mean(gen.propor.burnt[ta6 == "g__Tomentella",]$Proportion ) 
mean(gen.propor[ta6 == "g__Herpotrichia",]$Proportion ) ##0.2890
mean(gen.propor.burnt[ta6 == "g__Herpotrichia",]$Proportion ) 
mean(gen.propor[ta6 == "g__Mortierella",]$Proportion ) ##0.4100

mean(gen.propor.burnt[ta6 == "g__Mortierella",]$Proportion ) 
mean(gen.propor[ta6 == "g__Umbelopsis",]$Proportion ) ##3.98
mean(gen.propor.burnt[ta6 == "g__Umbelopsis",]$Proportion ) 
mean(gen.propor[ta6 == "g__Troposporella",]$Proportion ) ##0.038%
mean(gen.propor.burnt[ta6 == "g__Troposporella",]$Proportion )

#lets keep greater mean than 0.1 % across all samples 
df_fig.reduced<-df_fig[-which(is.na(df_fig$Genus)),]
df_fig.reduced<-filter(df_fig.reduced, Genus != "_Kalmusia" & Genus != "_Pleurophoma"&Genus !="_Acericola"&Genus !="_Dematiopleospora"&Genus !="_Juncaceicola"&Genus !="_Leptospora"&Genus !="_Phaeosphaeria"&Genus !="_Pseudoophiobolus"&Genus !="_Neocamarosporium"&Genus !="_Stemphylium"&Genus !="_Helicoma"&Genus !="_Trichophaea"&Genus !="_Coprinopsis"&Genus !="_Myxomphalia"&Genus !="_Pachylepyrium"&Genus !="_Cystobasidium"&Genus !="_Cotylidia"&Genus !="_Mastigobasidium"&Genus !="_Cystofilobasidium"&Genus !="_Itersonilia"&Genus !="_Udeniomyces"&Genus !="_Bullera"&Genus !="_Dioszegia"&Genus !="_Gelidatrema"&Genus !="_Papiliotrema"&Genus !="_Borealophlyctis"&Genus !="_Endoconidioma"&Genus !="_Rhodotorula"&Genus !="_Comoclathris"&Genus !="_Celosporium"&Genus !="_Filobasidium"&Genus !="_Sporobolomyces"&Genus !="_Rhodotorula"&Genus !="_Peziza"&Genus !="_Ramimonilia"&Genus !="_Curvibasidium"&Genus !="_Aureobasidium"&Genus !="_Fontanospora"&Genus !="_Sarcinomyces"&Genus !="_Mrakia"&Genus !="_Tausonia") # this is removing the burnt taxa
df_fig.reduced<- filter(df_fig.reduced, Genus != "_Phacidium" & Genus !="_Xenopolyscytalum" & Genus !="_Glarea" & Genus !="_Cryptosporiopsis" & Genus !="_Extremus"& Genus !="_Leucosporidium"& Genus !="_Athelia" & Genus !="_Phialocephala"& Genus !="_Chalara"& Genus !="_Meristemomyces"& Genus !="_Gremmenia"& Genus !="_Cladophialophora"& Genus !="_Scleropezicula"& Genus !="_Troposporella")

guild.df.abun.plot<-as.factor(c("Sp","Sp","P-Sp-Sy","Ye","Sp","Sy","Sy","Sp","Sy","Sp","Sp","P-Sp-E","E","Sp","E","P-Sp","E","E","E","E","Mo","Sp","P-Sp","P","Ye","Sp","P-Sp","P-Sp","P-Sp-Sy","P","Ye","Ye","P","Sp","Sp","E","Sp","Sp","Sp","P-Sp-Sy","Sp","Ye","Ye","Sp","Sp","Ye","Ye","Ye","Ye","Ye","P","P-Sp"))
length(guild.df.abun.plot$colour)
levels(guild.df.abun.plot)

guild.df.abun.plot<-as.data.frame(guild.df.abun.plot)

guild.df.abun.plot$colour<-ifelse(guild.df.abun.plot$guild.df.abun.plot == "E","olivedrab",ifelse(guild.df.abun.plot$guild.df.abun.plot == "Mo","orange",ifelse(guild.df.abun.plot$guild.df.abun.plot =="P","gray25",ifelse(guild.df.abun.plot$guild.df.abun.plot == "P-Sp","steelblue3",ifelse(guild.df.abun.plot$guild.df.abun.plot == "P-Sp-Sy","wheat4",ifelse(guild.df.abun.plot$guild.df.abun.plot == "Sp","darkorange3",ifelse(guild.df.abun.plot$guild.df.abun.plot == "P-Sp-E","yellow3",ifelse(guild.df.abun.plot$guild.df.abun.plot == "Sy","lightskyblue",ifelse(guild.df.abun.plot$guild.df.abun.plot == "Ye","lightpink1",as.character(guild.df.abun.plot$guild.df.abun.plot))))))))))

r = ggplot(data = df_fig.reduced, 
           aes(x = reorder(Genus,-`status(unburnt-burnt).x`), y = `status(unburnt-burnt).x`)) + 
  geom_bar(stat = "identity", width = 0.7, 
           position = position_dodge(width = 0.4), fill= guild.df.abun.plot$colour) +
  geom_errorbar(aes(ymin = `status(unburnt-burnt).x` - `status(unburnt-burnt).y`, ymax = `status(unburnt-burnt).x` + `status(unburnt-burnt).y`), width = 0.2,
                position = position_dodge(0.05), color = "black") + 
  labs(x = NULL, y = "Log fold change", 
       title = "Differential abundance between Unburnt and Burnt") + 
  theme_bw() + 
  theme(legend.position = "none",
        plot.title = element_text(hjust = 0.5),
        panel.grid.minor.y = element_blank(),
        axis.text.x = element_text(angle = 60, hjust = 1))
r

```

##bias-adjusted abundances burn
```{r}
dif.abun.stat<-dif.abun.status$samp_frac
dif.abun.stat[is.na(dif.abun.stat)]=0

log_obs_abn<-log(abundances(genus_data)+1)
log_abs_abn_adj<-t(t(log_obs_abn)-dif.abun.stat)
View(log_abs_abn_adj)
```

##bias-adjusted abundances date.2
```{r}
dif.abun.date.bias<-dif.abun.date$samp_frac #extract sampling fraction
dif.abun.date.bias[is.na(dif.abun.date.bias)]=0 # change from NA to 0 

log_obs_abn<-log(abundances(genus_data.nolp)+1) # log the abundances of genus data 
log_abs_abn_adj<-t(t(log_obs_abn)-dif.abun.date.bias) #subtract the log abundance from the smapling fraction
View(log_abs_abn_adj)
summary(log_abs_abn_adj)

#write into csv files
write_csv(as.data.frame(log_abs_abn_adj), "bias.adjusted.date.dif.abun.jul13.2022.csv")
rownames(otu_table(genus_data_burn))
write_csv(as.data.frame(rownames(otu_table(genus_data_burn))),"genus.names.jun1.2022.csv" )
```

##plotting bias-adjusted abundances over time for pyrophilous fungi 
```{r}
#optional read in the log_abs_abn_adj data frame
log_abs_abn_adj<-read_csv("bias.adjusted.date.dif.abun.jul13.2022.csv")
log_abs_abn_adj<-as.data.frame(log_abs_abn_adj)
rownames(log_abs_abn_adj)<-taxa_names(genus_data.nolp)
#create dataset with only the pyrophilous fungi 
log_abs_abn_adj<-t(log_abs_abn_adj) #transpose
only.pyro.log.abs.abn<-as.data.frame(subset(log_abs_abn_adj,select = c("g__Hormonema","g__Coniothyrium","g__Juncaceicola","g__Mycosphaerella",	"g__Aureobasidium",	"g__Alternaria",	"g__Fusicladium",	"g__Morchella",	"g__Peziza",	"g__Anthracobia","g__Geopyxis",	"g__Pyronema",	"g__Tricharina",	"g__Coniochaeta",	"g__Neurospora",	"g__Saitoella",	"g__Basidioascus",	"g__Filobasidium"	,"g__Naganishia",	"g__Holtermanniella"	,"g__Dioszegia","g__Vishniacozyma","g__Didymella","g__Spizellomyces","g__Calyptrozyma","g__Penicillium","g__Rhodosporidiobolus","g__Taphrina","g__Rhynchosporium","g__Cladosporium","g__Pholiota","g__Sporobolomyces","g__Warcupia","g__Perusta","g__Endoconidioma","g__Ramimonilia","g__Phaffia","g__Fontanospora","g__Sarcinomyces","g__Curvibasidium","g__Powellomyces","g__Pachylepyrium","g__Borealophlyctis","g__Rhodotorula","g__Comoclathris","g__Cotylidia","g__Cystofilobasidium","g__Myxomphalia","g__Coprinopsis","g__Phaeosphaeria","g__Papiliotrema","g__Mastigobasidium","g__Bullera","g__Itersonilia","g__Helicoma","g__Trichophaea","g__Gelidatrema","g__Pseudoophiobolus","g__Acericola","g__Neocamarosporium","g__Stemphylium","g__Cystobasidium","g__Dematiopleospora","g__Pleurophoma","g__Kalmusia","g__Leptospora","g__Udeniomyces"))) ## select for only the pyrophilous taxa above 0.05 %

only.pyro.log.abs.abn<-cbind(only.pyro.log.abs.abn,meta.data.microplots$Severity,meta.data.microplots$sample_date) #add the metadata
names(only.pyro.log.abs.abn)[names(only.pyro.log.abs.abn) == "meta.data.microplots$Severity"]<-"Severity"
names(only.pyro.log.abs.abn)[names(only.pyro.log.abs.abn) == "meta.data.microplots$sample_date"]<-"sample_date" ## rename columns
#remove the unburnt samples because I dont want there mean change over time influncing the ones in burnt
only.pyro.log.abs.abn<-only.pyro.log.abs.abn[only.pyro.log.abs.abn$Severity != "Unburnt",]

only.pyro.log.abs.abn.melt<-melt(only.pyro.log.abs.abn[,c(1:67,69)],id.vars = "sample_date",variable.name = "Genus") ##melt into data frame for visualizing, remove column 67 which is severity class
only.pyro.log.abs.abn.melt$value<-as.numeric(only.pyro.log.abs.abn.melt$value)#make value numeric

##create functions from http://www.cookbook-r.com/Graphs/Plotting_means_and_error_bars_(ggplot2)/
summarySEwithin <- function(data=NULL, measurevar, betweenvars=NULL, withinvars=NULL,
                            idvar=NULL, na.rm=FALSE, conf.interval=.95, .drop=TRUE) {

  # Ensure that the betweenvars and withinvars are factors
  factorvars <- vapply(data[, c(betweenvars, withinvars), drop=FALSE],
    FUN=is.factor, FUN.VALUE=logical(1))

  if (!all(factorvars)) {
    nonfactorvars <- names(factorvars)[!factorvars]
    message("Automatically converting the following non-factors to factors: ",
            paste(nonfactorvars, collapse = ", "))
    data[nonfactorvars] <- lapply(data[nonfactorvars], factor)
  }

  # Get the means from the un-normed data
  datac <- summarySE(data, measurevar, groupvars=c(betweenvars, withinvars),
                     na.rm=na.rm, conf.interval=conf.interval, .drop=.drop)

  # Drop all the unused columns (these will be calculated with normed data)
  datac$sd <- NULL
  datac$se <- NULL
  datac$ci <- NULL

  # Norm each subject's data
  ndata <- normDataWithin(data, idvar, measurevar, betweenvars, na.rm, .drop=.drop)

  # This is the name of the new column
  measurevar_n <- paste(measurevar, "_norm", sep="")

  # Collapse the normed data - now we can treat between and within vars the same
  ndatac <- summarySE(ndata, measurevar_n, groupvars=c(betweenvars, withinvars),
                      na.rm=na.rm, conf.interval=conf.interval, .drop=.drop)

  # Apply correction from Morey (2008) to the standard error and confidence interval
  #  Get the product of the number of conditions of within-S variables
  nWithinGroups    <- prod(vapply(ndatac[,withinvars, drop=FALSE], FUN=nlevels,
                           FUN.VALUE=numeric(1)))
  correctionFactor <- sqrt( nWithinGroups / (nWithinGroups-1) )

  # Apply the correction factor
  ndatac$sd <- ndatac$sd * correctionFactor
  ndatac$se <- ndatac$se * correctionFactor
  ndatac$ci <- ndatac$ci * correctionFactor

  # Combine the un-normed means with the normed results
  merge(datac, ndatac)
}
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
    library(plyr)

    # New version of length which can handle NA's: if na.rm==T, don't count them
    length2 <- function (x, na.rm=FALSE) {
        if (na.rm) sum(!is.na(x))
        else       length(x)
    }

    # This does the summary. For each group's data frame, return a vector with
    # N, mean, and sd
    datac <- ddply(data, groupvars, .drop=.drop,
      .fun = function(xx, col) {
        c(N    = length2(xx[[col]], na.rm=na.rm),
          mean = mean   (xx[[col]], na.rm=na.rm),
          sd   = sd     (xx[[col]], na.rm=na.rm)
        )
      },
      measurevar
    )

    # Rename the "mean" column    
    datac <- rename(datac, c("mean" = measurevar))

    datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean

    # Confidence interval multiplier for standard error
    # Calculate t-statistic for confidence interval: 
    # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
    ciMult <- qt(conf.interval/2 + .5, datac$N-1)
    datac$ci <- datac$se * ciMult

    return(datac)
}
normDataWithin <- function(data=NULL, idvar, measurevar, betweenvars=NULL,
                           na.rm=FALSE, .drop=TRUE) {
    library(plyr)

    # Measure var on left, idvar + between vars on right of formula.
    data.subjMean <- ddply(data, c(idvar, betweenvars), .drop=.drop,
     .fun = function(xx, col, na.rm) {
        c(subjMean = mean(xx[,col], na.rm=na.rm))
      },
      measurevar,
      na.rm
    )

    # Put the subject means with original data
    data <- merge(data, data.subjMean)

    # Get the normalized data in a new column
    measureNormedVar <- paste(measurevar, "_norm", sep="")
    data[,measureNormedVar] <- data[,measurevar] - data[,"subjMean"] +
                               mean(data[,measurevar], na.rm=na.rm)

    # Remove this subject mean column
    data$subjMean <- NULL

    return(data)
}

detach("package:dplyr", unload = TRUE)
##^note to run this piece of code need to uninstall dplyr! they clash 
se.mean.sum.pyro<-summarySEwithin(data=only.pyro.log.abs.abn.melt,measurevar = "value",withinvars = c("sample_date","Genus"),idvar = "Genus",conf.interval = 0.95)

##create ggplot2 by functional groups!
#first subset
se.mean.sum.pyro.sap.pat<-subset(se.mean.sum.pyro, subset = Genus == "g__Pyronema"|Genus =="g__Tricharina"|Genus =="g__Anthracobia"|Genus =="g__Neurospora"|Genus =="g__Hormonema"|Genus == "g_Peziza"|Genus == "g_Pholiota"|Genus == "g_Warcupia"|Genus == "g_Perusta"|Genus == "g_Ramimonilia"|Genus == "g_Fontanospora"|Genus =="g__Basidioascus"|Genus == "g__Rhynchosporium"|Genus =="g__Fusicladium"|Genus =="g__Mycosphaerella"|Genus == "g_Coniochaeta"|Genus == "g_Alternaria"|Genus == "g_Coniothyrium"|Genus == "g_Didymella"|Genus == "g_Spizellomyces"|Genus == "g_Cladosporium")

se.mean.sum.pyro.ye.mo<-subset(se.mean.sum.pyro, subset = Genus == "g__Saitoella"|Genus =="g__Vishniacozyma"|Genus =="g__Naganishia"|Genus =="g__Holtermanniella"|Genus =="g__Peziza"|Genus == "g__Alternaria"|Genus =="g__Calyptrozyma"|Genus =="g__Aureobasidium"|Genus =="g__Rhodosporidiobolus"|Genus =="g__Taphrina"|Genus =="g__Filobasidium"|Genus =="g__Hormonema"|Genus =="g__Sporobolomyces"|Genus =="g__Endoconidioma"|Genus =="g__Phaffia"|Genus =="g__Sarcinomyces"|Genus =="g__Curvibasidium"|Genus == "g_Penicillium")


se.mean.sum.pyro.myco<-subset(se.mean.sum.pyro, Genus =="g__Morchella"|Genus =="g__Geopyxis")

#then plot
#sapro
sapro.pyro<-ggplot(se.mean.sum.pyro.sap.pat, aes(x=sample_date, y=value, group = Genus,shape=Genus,color=Genus))+geom_line() +geom_point(size=2)+labs(title="",x="Sample Month", y = "Log(bias-adjusted abundances)")+theme_classic()+geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1)+scale_y_continuous(limits=c(-0.2,10))
#+scale_color_manual(values = c("tan1","tan2","tan3","tan1","tan2","tan3"))

myco.pyro<-ggplot(se.mean.sum.pyro.myco, aes(x=sample_date, y=value, group = Genus, color=Genus,shape=Genus))+geom_line() +geom_point(size=2)+labs(title="",x="Sample Month", y = "")+theme_classic()+geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1)+scale_y_continuous(limits=c(-0.2,10))
#+scale_color_manual(values = c("indianred2","darkolivegreen"))

ye.mo.pyro<-ggplot(se.mean.sum.pyro.ye.mo, aes(x=sample_date, y=value, group = Genus, color=Genus,shape=Genus))+geom_line() +geom_point(size=2)+labs(title="",x="Sample Month", y = "")+theme_classic()+geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1)+scale_y_continuous(limits=c(-0.2,6))#+scale_shape_manual(values = c(0,1,2,4,5,6,7,8,9,10,11))
#+scale_color_manual(values = c("gray72","gray68","gray64","gray60","gray56","gray52","gray48","gray44","gray40","gray42"))
View(se.mean.sum.pyro.unkn)

ggarrange(sapro.pyro,myco.pyro,ye.mo.pyro,ncol = 2,nrow = 2)

se.mean.sum.pyro
write_csv(se.mean.sum.pyro, "mean.se.sum.pyro.aug27.2022.csv")
```

#cross reference whether or not pyro taxa are present in the unburnt
```{r}
unburnt.phylo<-subset_samples(master.phylo.2.lp, burn.status == "Unburnt")

#cleaar out absent taxa
alltaxa.unburnt<-otu_table(unburnt.phylo)
goodtaxa.unburnt<-alltaxa.unburnt[-which(rowSums(otu_table(unburnt.phylo)) == 0),]
goodtaxa.unburnt<-otu_table(goodtaxa.unburnt,taxa_are_rows = TRUE)
taxa.good.unburnt<-as.data.frame(rownames(goodtaxa.unburnt))
colnames(taxa.good.unburnt)<-"full.taxa"
taxa.good.unburnt<-separate(taxa.good.unburnt,col= "full.taxa",into = c("Kingdom","Phylum","Class","Order","Family","Genus","Species"),sep = ";")
good.taxonomy.unburnt<-tax_table(taxa.good.unburnt)
taxa_names(good.taxonomy.unburnt)<-rownames(goodtaxa.unburnt)
sd.unburnt<-sample_data(unburnt.phylo)
unburnt.phylo<-merge_phyloseq(sd.unburnt,goodtaxa.unburnt,good.taxonomy.unburnt)
unburnt.phylo

genus_data_unburnt<-aggregate_taxa(unburnt.phylo,"ta6")

pyro.v.unburnt<-setdiff(row.names(dif.abun.status.nolp$res$beta),taxa_names(genus_data_unburnt))
pyro.v.unburnt

#unburnt sept 
sept.ub.phylo<-subset_samples(unburnt.phylo, sample_date == "September")
alltaxa.sept.ub<-otu_table(sept.ub.phylo)
goodtaxa.sept.ub<-alltaxa.sept.ub[-which(rowSums(otu_table(sept.ub.phylo)) == 0),]
goodtaxa.sept.ub<-otu_table(goodtaxa.sept.ub,taxa_are_rows = TRUE)
taxa.good.sept.ub<-as.data.frame(rownames(goodtaxa.sept.ub))
colnames(taxa.good.sept.ub)<-"full.taxa"
taxa.good.sept.ub<-separate(taxa.good.sept.ub,col= "full.taxa",into = c("Kingdom","Phylum","Class","Order","Family","Genus","Species"),sep = ";")
good.taxonomy.sept.ub<-tax_table(taxa.good.sept.ub)
taxa_names(good.taxonomy.sept.ub)<-rownames(goodtaxa.sept.ub)
sd.sept.ub<-sample_data(sept.ub.phylo)
sept.ub.phylo<-merge_phyloseq(sd.sept.ub,goodtaxa.sept.ub,good.taxonomy.sept.ub)
sept.ub.phylo

#burnt sept 
burnt.phylo<-subset_samples(master.phylo.2.lp, burn.status == "Burnt")

sept.b.phylo<-subset_samples(burnt.phylo, sample_date == "September")
alltaxa.sept.b<-otu_table(sept.b.phylo)
goodtaxa.sept.b<-alltaxa.sept.b[-which(rowSums(otu_table(sept.b.phylo)) == 0),]
goodtaxa.sept.b<-otu_table(goodtaxa.sept.b,taxa_are_rows = TRUE)
taxa.good.sept.b<-as.data.frame(rownames(goodtaxa.sept.b))
colnames(taxa.good.sept.b)<-"full.taxa"
taxa.good.sept.b<-separate(taxa.good.sept.b,col= "full.taxa",into = c("Kingdom","Phylum","Class","Order","Family","Genus","Species"),sep = ";")
good.taxonomy.sept.b<-tax_table(taxa.good.sept.b)
taxa_names(good.taxonomy.sept.b)<-rownames(goodtaxa.sept.b)
sd.sept.b<-sample_data(sept.b.phylo)
sept.b.phylo<-merge_phyloseq(sd.sept.b,goodtaxa.sept.b,good.taxonomy.sept.b)
sept.b.phylo

#october 
oct.ub.phylo<-subset_samples(unburnt.phylo, sample_date == "October")
alltaxa.oct.ub<-otu_table(oct.ub.phylo)
goodtaxa.oct.ub<-alltaxa.oct.ub[-which(rowSums(otu_table(oct.ub.phylo)) == 0),]
goodtaxa.oct.ub<-otu_table(goodtaxa.oct.ub,taxa_are_rows = TRUE)
taxa.good.oct.ub<-as.data.frame(rownames(goodtaxa.oct.ub))
colnames(taxa.good.oct.ub)<-"full.taxa"
taxa.good.oct.ub<-separate(taxa.good.oct.ub,col= "full.taxa",into = c("Kingdom","Phylum","Class","Order","Family","Genus","Species"),sep = ";")
good.taxonomy.oct.ub<-tax_table(taxa.good.oct.ub)
taxa_names(good.taxonomy.oct.ub)<-rownames(goodtaxa.oct.ub)
sd.oct.ub<-sample_data(oct.ub.phylo)
oct.ub.phylo<-merge_phyloseq(sd.oct.ub,goodtaxa.oct.ub,good.taxonomy.oct.ub)
oct.ub.phylo

#burnt 
oct.b.phylo<-subset_samples(burnt.phylo, sample_date == "October")
alltaxa.oct.b<-otu_table(oct.b.phylo)
goodtaxa.oct.b<-alltaxa.oct.b[-which(rowSums(otu_table(oct.b.phylo)) == 0),]
goodtaxa.oct.b<-otu_table(goodtaxa.oct.b,taxa_are_rows = TRUE)
taxa.good.oct.b<-as.data.frame(rownames(goodtaxa.oct.b))
colnames(taxa.good.oct.b)<-"full.taxa"
taxa.good.oct.b<-separate(taxa.good.oct.b,col= "full.taxa",into = c("Kingdom","Phylum","Class","Order","Family","Genus","Species"),sep = ";")
good.taxonomy.oct.b<-tax_table(taxa.good.oct.b)
taxa_names(good.taxonomy.oct.b)<-rownames(goodtaxa.oct.b)
sd.oct.b<-sample_data(oct.b.phylo)
oct.b.phylo<-merge_phyloseq(sd.oct.b,goodtaxa.oct.b,good.taxonomy.oct.b)
oct.b.phylo
```


##DA with DEseq2 burnt versus unburnt 
```{r}
# create metadata variable where severe, med-severe, medium, surface all = burnt 
meta.data.microplots.fall$burn.status<-ifelse(meta.data.microplots.fall$Severity=="Severe","Burnt",if_else(meta.data.microplots.fall$Severity=="Medium-Severe","Burnt",if_else(meta.data.microplots.fall$Severity=="Medium","Burnt",if_else(meta.data.microplots.fall$Severity=="Low","Burnt",if_else(meta.data.microplots.fall$Severity=="Unburnt","Unburnt",NA_character_)))))

meta.data.microplots.spring$burn.status<-ifelse(meta.data.microplots.spring$Severity=="LP-Severe","Burnt",ifelse(meta.data.microplots.spring$Severity=="Severe","Burnt",if_else(meta.data.microplots.spring$Severity=="Medium-Severe","Burnt",if_else(meta.data.microplots.spring$Severity=="Medium","Burnt",if_else(meta.data.microplots.spring$Severity=="Low","Burnt",if_else(meta.data.microplots.spring$Severity=="Unburnt","Unburnt",NA_character_))))))


##read in new meta.data to phyloseq
phylo.sample.meta.fall.status<-sample_data(meta.data.microplots.fall)
phylo.sample.meta.spring.status<-sample_data(meta.data.microplots.spring)
sample_names(phylo.sample.meta.spring.status)<-sample_names(phylo.sample.meta.spring.microplots)
sample_names(phylo.sample.meta.fall.status)<-sample_names(phylo.sample.meta.fall.microplots)
phylo.fall.2<-phyloseq(microplot.taxa.fall,phylo.taxonomy.fall,phylo.sample.meta.fall.status)
phylo.spring.2<-phyloseq(microplot.taxa.spring,phylo.taxonomy.spring,phylo.sample.meta.spring.status)

master.phylo<-merge_phyloseq(phylo.spring.2,phylo.fall.2)
sample_data(master.phylo)

##remove taxa with zero sums
pop_taxa = function(physeq, badTaxa){
  allTaxa = taxa_names(physeq)
  allTaxa <- allTaxa[!(allTaxa %in% badTaxa)]
  return(prune_taxa(allTaxa, physeq))
}
which(rowSums(otu_table(master.phylo)) == 0)
badTaxa<-c("k__Fungi;p__Ascomycota;c__Eurotiomycetes;o__Eurotiales;f__Trichocomaceae;g__Talaromyces;s__Talaromyces_marneffei","k__Fungi;p__Ascomycota;c__Lecanoromycetes;o__Lecanorales;f__Cladoniaceae;g__Cladonia;s__Cladonia_stipitata","k__Fungi;p__Ascomycota;c__Pezizomycetes;o__Pezizales;f__Morchellaceae;g__Morchella;__" ,"k__Fungi;p__Ascomycota;c__Saccharomycetes;o__Saccharomycetales;f__Pichiaceae;g__Nakazawaea;s__Nakazawaea_holstii" ,"k__Fungi;p__Ascomycota;c__Saccharomycetes;o__Saccharomycetales;f__Saccharomycetales_fam_Incertae_sedis;g__Candida;s__Candida_odintsovae" ,"k__Fungi;p__Ascomycota;c__Sordariomycetes;o__Diaporthales;f__Valsaceae;g__Cytospora;s__Cytospora_prunicola" ,"k__Fungi;p__Ascomycota;c__Sordariomycetes;o__Hypocreales;f__Clavicipitaceae;g__Metarhizium;s__Metarhizium_marquandii" ,"k__Fungi;p__Ascomycota;c__Sordariomycetes;o__Hypocreales;f__Nectriaceae;g__Gibberella;s__Gibberella_baccata","k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Agaricales;f__Entolomataceae;g__unidentified;s__unidentified" ,"k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Agaricales;f__Inocybaceae;g__Inocybe;s__Inocybe_subporospora" ,"k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Agaricales;f__Tricholomataceae;g__Flagelloscypha;s__Flagelloscypha_minutissima" ,"k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Russulales;f__Russulaceae;g__Russula;s__Russula_olivobrunnea","k__Fungi;p__Mucoromycota;c__Mucoromycetes;o__Mucorales;f__Mucoraceae;g__Mucor;s__Mucor_circinelloides")
master.phylo.deseq<-pop_taxa(master.phylo,badTaxa=badTaxa)
which(rowSums(otu_table(master.phylo.deseq)) == 0)

##perform analysis
dif.ab.deseq<-phyloseq_to_deseq2(master.phylo.deseq, ~burn.status)
dif.ab.deseq<-DESeq2::DESeq(dif.ab.deseq,test = "Wald",fitType = "parametric",sfType = "poscounts")

#investigate results
res<-DESeq2::results(dif.ab.deseq,cooksCutoff=FALSE)
DESeq2::resultsNames(dif.ab.deseq)
resLFC<-DESeq2::lfcShrink(dif.ab.deseq,coef = "burn.status_Unburnt_vs_Burnt",type= "apeglm")
alpha<-0.01
sigtab<-res[which(res$padj < alpha),]
sigtab.lfc<-resLFC[which(resLFC$padj < alpha),]
sigtab<-cbind(as(sigtab, "data.frame"),as(tax_table(master.phylo.deseq)[rownames(sigtab),],"matrix"))
sigtab.lfc<-cbind(as(sigtab.lfc, "data.frame"),as(tax_table(master.phylo.deseq)[rownames(sigtab.lfc),],"matrix"))
head(sigtab)
head(res)

#visualize

DESeq2::plotMA(res,ylim=c(-15,15))
DESeq2::plotMA(resLFC,ylim=c(-15,15))

DESeq2::plotCounts(dif.ab.deseq,gene = which.min(res$padj), intgroup = "burn.status")


theme_set(theme_bw())
scale_fill_discrete<-function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)}
  
x<-tapply(sigtab$log2FoldChange,sigtab$ta6, function(x) max(x))
x<-sort(x,TRUE)

sigtab$ta6<-factor(as.character(sigtab$ta6),levels = names(x))

sigtab<-sigtab[-which(sigtab$ta6 == "__"),]

ggplot(sigtab,aes(x=ta6,y=log2FoldChange, color= color.sigtab))+geom_point(size=3)+theme(axis.text.x = element_text(angle=-90,hjust = 0,vjust = 0.5),legend.position = "none")+geom_abline(intercept = 0,slope = 0)+scale_colour_manual(values = c("black","blue4","gray","darkolivegreen","darkorange3","darkmagenta","darkred","darkgoldenrod1"))+labs(x="",y="log2FoldChange")

x<-tapply(sigtab.lfc$log2FoldChange,sigtab.lfc$ta6, function(x) max(x))
x<-sort(x,TRUE)

sigtab.lfc$ta6<-factor(as.character(sigtab.lfc$ta6),levels = names(x))

sigtab.lfc<-sigtab.lfc[-which(sigtab.lfc$ta6 == "__"),]

ggplot(sigtab.lfc,aes(x=ta6,y=log2FoldChange, color= color.sigtab))+geom_point(size=3)+theme(axis.text.x = element_text(angle=-90,hjust = 0,vjust = 0.5),legend.position = "none")+geom_abline(intercept = 0,slope = 0)+scale_colour_manual(values = c("black","blue4","gray","darkolivegreen","darkorange3","darkmagenta","darkred","darkgoldenrod1"))+labs(x="",y="log2FoldChange")


#plot with functional guilds as colours 

sigtab$ta6

color.sigtab<-c("gray","gray","gray","gray","black","gray","green","yellow","yellow","gray","blue","blue","gray","gray","green","green","green","yellow","gray","gray","black","gray","gray","yellow","gray","gray","yellow","gray","gray","gray","yellow","gray","purple","purple","yellow","yellow","yellow","yellow","yellow","purple","yellow","yellow","black","yellow","gray","black","gray","orange","gray","yellow","red","red","yellow","yellow","yellow","red","yellow","red","gray","gray","green","green","green","yellow","yellow","gray","black","yellow","red","orange","red","yellow","gray","yellow","yellow","yellow","blue","blue","yellow","red","red","red","gray","red","gray","red","red","red","gray","gray","yellow","gray","gray","blue","blue","yellow","gray","gray","gray","gray","yellow","gray","gray","gray","gray","gray","gray","gray","gray","gray","gray","gray","yellow","black","blue","blue","gray","orange","gray")
length(color.sigtab)
length(sigtab$ta6)
```


